daemons/controld/controld_remote_ra.c

```diff
@@ -219,6 +219,27 @@ purge_remote_node_attrs(int call_opt, const char *node_name)
         section = controld_section_lrm;
     }
 
+    /* Check the uptime of connection_rsc.  If it hasn't been running long
+     * enough, set purge=true regardless of what we saw above.  "Long enough"
+     * means it started running earlier than the timestamp when we noticed
+     * it went away in the first place.
+     */
+    if (connection_rsc && connection_rsc->conn) {
+        remote_ra_data_t *ra_data = connection_rsc->remote_ra_data;
+        lrmd_t *lrm = connection_rsc->conn;
+        time_t uptime = lrmd__uptime(lrm);
+        time_t now = time(NULL);
+
+        /* Add 20s of fuzziness to give corosync a while to notice the
+         * remote host is gone.
+         */
+        if (ra_data == NULL || (ra_data && uptime < 20 + now - ra_data->when_host_lost)) {
+            purge = true;
+        }
+    } else {
+        purge = true;
+    }
```

Comment:
I was imagining remote_ra_purge_attrs as a function that only returned the state of the flag, in the same like as remote_ra_is_in_maintenance or remote_ra_controlling_guest just above it, in case someone else eventually needed to know that information.

Reply:
That's fine, it could be an intermediary function like should_purge_attributes(). The idea is that if we don't do that, then we may need to adjust section after this block

Reply:
Updated.